File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:

        read_idx = 1
        write_idx =0
        unique_cnt = 1

        while read_idx < len(nums):

            if nums[read_idx] == nums[read_idx-1]:
                read_idx+=1
                continue
            else:
                nums[write_idx+1] = nums[read_idx]
                read_idx+=1
                write_idx+=1
                unique_cnt+=1  
                
        return unique_cnt          
        

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {

        int read_idx = 1;
        int write_idx = 0;
        int unique_cnt = 1;
        int nums_len = nums.length;

        while(read_idx <  nums_len)
        {
            if(nums[read_idx] == nums[read_idx-1])
            {
                read_idx++;
                continue;
            }
            else
            {
                nums[write_idx+1] = nums[read_idx];
                write_idx++;
                read_idx++;
                unique_cnt++;
            }
        }
        return unique_cnt; 
    }
}